sqlconnect/internal/postgres/config.go

```diff
@@ -29,14 +33,28 @@ func (c Config) ConnectionString() string {
 	if c.SSLMode != "" {
 		sslMode = c.SSLMode
 	}
-	return fmt.Sprintf("host=%s port=%d dbname=%s user=%s password=%s sslmode=%s", c.Host, c.Port, c.DBName, c.User, c.Password, sslMode)
+	dsn := url.URL{
+		Scheme: DatabaseType,
+		User:   url.UserPassword(c.User, c.Password),
+		Host:   fmt.Sprintf("%s:%d", c.Host, c.Port),
+		Path:   c.DBName,
+	}
+	values := dsn.Query()
+	values.Set("sslmode", sslMode)
+	dsn.RawQuery = values.Encode()
+	return dsn.String()
```

Comment:
**Note:** using a dsn for postgres connection string, for properly escaping username/passwords